Sources/Apollo/ApolloStore.swift

```diff
@@ -188,17 +188,39 @@ public final class ApolloStore {
   }
   
   public final class ReadWriteTransaction: ReadTransaction {
-    public func write<Query: GraphQLQuery>(object: JSONObject, forQuery query: Query) throws {
-      try write(object: object, forSelectionSet: Query.selectionSet, withKey: rootKey(forOperation: query), variables: query.variables)
+    public func update<Query: GraphQLQuery>(query: Query, _ body: (inout Query.Data) throws -> Void) throws {
+      var data = try read(query: query)
+      try body(&data)
+      try write(data: data, forQuery: query)
     }
     
-    public func write<Fragment: GraphQLFragment>(object: JSONObject, forFragment fragment: Fragment.Type, withKey key: CacheKey, variables: GraphQLMap? = nil) throws {
-      try write(object: object, forSelectionSet: fragment.selectionSet, withKey: key, variables: variables)
+    public func updateFragment<Fragment: GraphQLFragment>(ofType fragmentType: Fragment.Type, withKey key: CacheKey, variables: GraphQLMap? = nil, _ body: (inout Fragment) throws -> Void) throws {
+      var fragment = try readFragment(ofType: fragmentType, withKey: key, variables: variables)
+      try body(&fragment)
+      try write(fragment: fragment, withKey: key, variables: variables)
     }
     
-    func write(object: JSONObject, forSelectionSet selectionSet: [Selection], withKey key: CacheKey, variables: GraphQLMap?) throws {
+    public func updateSelectionSet<SelectionSet: GraphQLSelectionSet>(ofType selectionSetType: SelectionSet.Type, withKey key: CacheKey, variables: GraphQLMap? = nil, _ body: (inout SelectionSet) throws -> Void) throws {
```

Comment:
Yeah, I'm not a 100% sure if this is needed. But the idea is that you don't always want to define a fragment just to read or update data. So you can update a particular `AllPostsQuery.Data.Post` directly for instance.